### PR TITLE
Radar workflow: some fix for stack processing

### PR DESCRIPTION
**Pull request by vbrancat**
While I was running the range/Doppler workflow for a stack of burst I noticed that we are not able to correctly pick up the radar grid for the burst to process.

This PR:
1. Isolates and picks up the correct radar grid for the burst to process
2. Adds date and burst ID to the output coregistered SLC.

### CODE REVIEW

**Review by hfattahi — APPROVED** [review]
LGTM!

**Comment by vbrancat** [comment]
For the record (since I did not write the correct message while committing), the latest commit should:

1. Modify the azimuth carrier to have range/azimuth vectors as an input (instead of indices). This fixes the issue we had in forming the interferogram with the radar coordinates workflow. Now, the interferogram looks nice
2. Assign a direct path to the radar grid. Easier for the stack processor. This requires to give the direct path to the `topo.vrt` inside the workflow runconfig

File: src/compass/utils/runconfig.py [hunk 1]
```diff
@@ -241,7 +241,7 @@ def runconfig_to_bursts(cfg: SimpleNamespace) -> list[Sentinel1BurstSlc]:
     return bursts
 
 
-def get_ref_radar_grid_info(ref_path, burst_id):
+def get_ref_radar_grid_info(ref_path, burst_id, date_str):
```
[the remark this answers is not in the document]
Comment (vbrancat) [reply src/compass/utils/runconfig.py]: I also removed the check on multiple radar grids as this logic is not needed anymore

File: src/compass/s1_resample.py [hunk 2]
```diff
@@ -84,7 +84,7 @@ def run(cfg: dict):
         original_raster = isce3.io.Raster(sec_burst_path)
 
         # Prepare resamled SLC as raster object
-        coreg_burst_path = f'{burst_output_path}/{pol}.slc'
+        coreg_burst_path = f'{burst_output_path}/{burst_id}_{date_str}_{pol}.slc'
```
[the remark this answers is not in the document]
Comment (LiangJYu) [reply src/compass/s1_resample.py:87]: I am referring to exactly that.

**Review by LiangJYu — COMMENTED** [review]
LGTM. Thanks for the fast turnaround!

**Review by LiangJYu — APPROVED** [review]

**Pull request merged**